### PR TITLE
Boos 218/clean up

### PR DESCRIPTION
**Pull request by koutsilisk**
- Implement __repr__ function to all models :white_check_mark: 
- Update invalid value tests with {invalid_field: "invalid value"} :white_check_mark: 
- At services, make blacklist a constant :white_check_mark: 
- Be consistent in the code: testing is done on an unwrapped req dict then on individual argument :white_check_mark: 
- Research if using Dict() is great or not and refactor accordingly (seen in tests) :white_check_mark: 
- Implement Model.__eq()__ for easy comparison in tests :white_check_mark: 
- Update marshmallow schema :white_check_mark:

### CODE REVIEW

**Comment by koutsilisk** [comment]
@joaopcartier I will need your help to resolve the frontend conflicts :/

**Comment by joaopcartier** [comment]
> @joaopcartier I will need your help to resolve the frontend conflicts :/

Just resolved it

File: backend/project/api/models/user_role.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+from sqlalchemy.sql.functions import user
```
Comment (Chichilele) [inline backend/project/api/models/user_role.py:1]: What's this for? looks like an auto import

File: backend/project/api/routes/roles_routes.py [hunk 2]
```diff
@@ -45,10 +47,16 @@ def get_role_by_id(role_id: int):
 @api.route("/roles/<int:role_id>", methods=["PATCH"])
 def update_role(role_id: int):
     request_data = request.json
+    # TODO error handling
     if not request_data:
+        return jsonify(message="Payload required"), 400
```
Comment (Chichilele) [inline backend/project/api/routes/roles_routes.py:52]: I would even remove that to reduce code as much as possible. It'll be handled by the deserializer

File: backend/project/api/schemas/group.py [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
         model = Group
+        dump_only = ("id", "created_at", "updated_at")
 
-    id = ma.auto_field()
-    groupname = ma.auto_field()
-    description = ma.auto_field()
-    created_at = ma.auto_field()
-    updated_at = ma.auto_field()
+    id = fields.Int()
+    groupname = fields.Str(required=True)
```
Comment (Chichilele) [inline backend/project/api/schemas/group.py:12]: to update to name sorry!

File: backend/project/api/services/roles.py [hunk 4]
```diff
@@ -67,12 +71,11 @@ def update_role(self, role_id: int, **kwargs) -> Role:
         Returns:
             Role: Role instance with updated fields
         """
-        blacklist = ["created_at", "updated_at", "id"]
         role = self.get_role_by_id(role_id)
         if not role:
             return None
         for attr, value in kwargs.items():
-            if attr in blacklist or not hasattr(role, attr):
+            if attr in self.BLACKLIST_FIELDS or not hasattr(role, attr):
```
Comment (Chichilele) [inline backend/project/api/services/roles.py:78]: cleaner indeed. Could even be a contstant somewhere

File: backend/project/tests/schemas/test_group_schema.py [hunk 5]
```diff
@@ -4,7 +4,7 @@
 
 class TestGroupSchema:
     @pytest.fixture
-    def serialized_group(self):
+    def serialized_group_dump(self):
```
Comment (Chichilele) [inline backend/project/tests/schemas/test_group_schema.py:7]: ooh good one! They're different now

File: frontend/src/components/role/roleSlice.ts [hunk 6]
```diff
@@ -121,6 +121,13 @@ export interface RoleState {
   );
 
 
+  export const requestRole = createAsyncThunk(
+    "user/requestUser",
+    async (id: number) => {
+      return await getRoleById(id);
+    }
+  )
+
```
Comment (Chichilele) [inline frontend/src/components/role/roleSlice.ts:130]: Have you ckecked with @joaopcartier ?

**Review by Chichilele — APPROVED** [review]
good stuff. Couple of questions for my education :-)

File: frontend/src/components/role/roleSlice.ts (hunk 6 above)
Comment (Chichilele) [reply frontend/src/components/role/roleSlice.ts:130]: yes, just saw the comments above

File: backend/project/api/models/user_role.py (hunk 1 above)
Comment (koutsilisk) [reply backend/project/api/models/user_role.py:1]: auto import, deleted

File: backend/project/api/routes/roles_routes.py (hunk 2 above)
Comment (koutsilisk) [reply backend/project/api/routes/roles_routes.py:52]: its not handled because of the partial=true

Comment (koutsilisk) [reply backend/project/api/routes/roles_routes.py:52]: so an empty request is considered valid

File: backend/project/api/services/roles.py (hunk 4 above)
Comment (koutsilisk) [reply backend/project/api/services/roles.py:78]: somewhere 💯 :p

**Pull request merged**